Functions/Shortcuts/Wine/script.js

```diff
@@ -34,7 +34,7 @@ var WineShortcut = function () {
 
     that.miniature = function(miniature) {
       if(isArray(miniature)) {
-          var application = that._appsManager.getApplication(miniature);
+          var application = that._appsManager.cachedRepository().getApplication(miniature);
```

Comment:
I know what you mean. 
My problem with the current approach is, that we're mixing direct returns and callbacks. 
We use callbacks for the population of the "Apps" tab, while we're using direct returns for everything else.
To change this we should standardize the access to the applications, categories and repositories.